src/libc/musl/string.c

```diff
@@ -0,0 +1,63 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+
+/* The header string.h overrides strerror_r with strerror_r_gnu, hence we need to undef it here. */
+#undef strerror_r
+
+char* strerror_r_gnu(int errnum, char *buf, size_t buflen) {
+        int saved_errno = errno;
+        const char *ret;
+
+        /* musl provides spurious catchall error message "No error information" for unknown errno
+         * (including errno == 0). Let's patch it to glibc style. */
+
+        if (errnum == 0) {
+                ret = "Success";
+                goto finalize;
+        }
+
+        ret = strerror(errnum);
+        if (ret != strerror(0)) /* Yay, we got something unique. */
+                goto finalize;
```

Comment:
No, please don't use `strerror`. This is an implementation for `strerror_r`, which is specifically supposed to be reentrant. Probably, for the particular case of 0 the function would return a fixed allocation, but this is not guaranteed by the standard and it's best to avoid any issues.